src/main/java/com/movie_collection/gui/controllers/BaseController.java

```diff
@@ -1,35 +1,139 @@
 import com.movie_collection.gui.controllers.abstractController.RootController;
 import com.movie_collection.gui.controllers.controllerFactory.IControllerFactory;
+import com.movie_collection.gui.models.ICategoryModel;
 import javafx.event.ActionEvent;
 import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
 import javafx.scene.Parent;
 import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.layout.HBox;
 import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
 import javafx.stage.Modality;
 import javafx.stage.Stage;
 
 import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Base controller navigation with switchable content
  * Serves as a root for all the action :)
  * injects the root controller
  */
-public class BaseController extends RootController {
+public class BaseController extends RootController implements Initializable {
 
+    @FXML
+    private ScrollPane scroll_pane;
     @FXML
     private StackPane app_content;
 
     @Inject
     IControllerFactory rootController;
 
     @Inject
-    public BaseController(IControllerFactory rootController) {
+    ICategoryModel categoryModel;
+
+    @Inject
+    public BaseController(IControllerFactory rootController,ICategoryModel categoryModel) {
         this.rootController = rootController;
+        this.categoryModel = categoryModel;
+    }
+
+    //  TODO: an empty constructor that is always created
+    // whenever we specify but in this case it wa
+    // s doing some not good stuff with injection
+    // but lets look into the di AND MODULE-INFO LATER
+    public BaseController() {
+    }
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        setCategoriesScrollPane(categoryModel.getAllCategories());
+    }
+
+    /**
+     * void method that invokes scroll pane to clean content first
+     * and then set it back to all categories
+     */
+    public void refreshScrollPane(){
+        if(scroll_pane.getContent() != null){
+            scroll_pane.setContent(null);
+            setCategoriesScrollPane(categoryModel.getAllCategories());
+        }
+    }
+
+    /**
+     * method to set all the categories from List<Category> that uses Linked hashmap to store two buttons that will be then
+     * one by one added into the empty scroll pane
+     * if no categories are provided stack pane is filled with Label that notifies the user about not having any category listed yet
+     * LinkedHashMap is added here due to the reason that whenever new Button is added it is automatickly put as the last and keps the order
+     * TODO: Maybe all of the body can be exctracted into separated class since it look so hoooge
+     * @param allCategories list of all categories
+     */
+    private void setCategoriesScrollPane(List<Category> allCategories) {
```

Comment:
hope this piece of code makes sense otherwise ofc we can go later and check its full implementation and what it does execution line by line

Reply:
maybe calling the buttons as btn  and btn2 is not good naming

Reply:
yeah use proper name like deleteButton / categoryButton that should be fine